ocs_ci/deployment/vmware.py

```diff
@@ -24,23 +32,306 @@ def __init__(self):
+            worker_ignition = read_file_as_str(f"{worker_ignition_path}")
+            config.ENV_DATA['compute_ignition'] = worker_ignition
+
+            cluster_domain = f"{config.ENV_DATA.get('cluster_name')}.{config.ENV_DATA.get('base_domain')}"
+            config.ENV_DATA['cluster_domain'] = cluster_domain
+
+            # generate terraform variables from template
+            logger.info("Generating terraform variables")
+            _templating = Templating()
+            terraform_var_template = "terraform.tfvars.j2"
+            terraform_var_template_path = os.path.join(
+                "ocp-deployment", terraform_var_template
+            )
+            terraform_config_str = _templating.render_template(
+                terraform_var_template_path, config.ENV_DATA
+            )
+
+            terraform_var_yaml = os.path.join(self.cluster_path, "terraform.tfvars.yaml")
+            with open(terraform_var_yaml, "w") as f:
+                f.write(terraform_config_str)
+            self.terraform_var = self.convert_yaml2tfvars(terraform_var_yaml)
+
+            # update gateway and DNS
+            installer_ignition = f"{self.upi_repo_path}/upi/vsphere/machine/ignition.tf"
+            if config.ENV_DATA.get('gateway'):
+                replace_content_in_file(
+                    installer_ignition,
+                    '${cidrhost(var.machine_cidr,1)}',
+                    f"{config.ENV_DATA.get('gateway')}"
+                )
+
+            if config.ENV_DATA.get('dns'):
+                replace_content_in_file(
+                    installer_ignition,
+                    constants.INSTALLER_DEFAULT_DNS,
+                    f"{config.ENV_DATA.get('dns')}"
+                )
+
+            # update the zone in route
+            if config.ENV_DATA.get('region'):
+                route53 = f"{self.upi_repo_path}/upi/vsphere/route53/main.tf"
+                def_zone = 'provider "aws" { region = "%s" } \n' % config.ENV_DATA.get('region')
+                replace_content_in_file(route53, "xyz", def_zone)
+
+            # increase memory
+            if config.ENV_DATA.get('memory'):
+                machine_conf = f"{self.upi_repo_path}/upi/vsphere/machine/main.tf"
+                replace_content_in_file(
+                    machine_conf,
+                    constants.INSTALLER_DEFAULT_MEMORY,
+                    config.ENV_DATA.get('memory')
+                )
+
+        def convert_yaml2tfvars(self, yaml):
+            """
+            Converts yaml file to tfvars. It creates the tfvars with the
+            same filename in the required format which is used for deployment.
+            Args:
+                yaml (str): File path to yaml
+            Returns:
+                str: File path to tfvars
+            """
+            data = load_yaml(yaml)
+            tfvars_file = f"{yaml.split('.')[0]}.tfvars"
+            fd = open(tfvars_file, "w+")
```

Comment:
I would suggest to use `with` construction to open the file (e.g. like [here](https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/deployment/ocp.py#L62)).

Reply:
Done